analyses/data-pre-release-qc/01-data-harmonization-qc.Rmd

```diff
@@ -0,0 +1,752 @@
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = beta_values)
+
+# m-values matrix
+m_values <-
+  tibble::tibble(m_values =
+    readr::read_rds(file.path(data_dir, "methyl-m-values.rds")) %>%
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = m_values)
+
+# cn-values matrix
+cn_values <-
+  tibble::tibble(cn_values =
+    readr::read_rds(file.path(data_dir, "methyl-cn-values.rds")) %>%
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = cn_values)
+
+# merge and list differences
+probe_id_diffs <- beta_values %>%
+  dplyr::full_join(m_values, by = "probe_id") %>%
+  dplyr::full_join(cn_values, by = "probe_id") %>%
+  dplyr::select(-probe_id) %>%
+  dplyr::filter(is.na(beta_values) | is.na(m_values) | is.na(cn_values)) %>%
+  dplyr::distinct()
+
+# check any differences in probe IDs among methyl matrices
+if (length(rownames(probe_id_diffs)) > 0) {
+  output_file <- "probes_not_in_all_methyl_matrices.tsv"
+  gene_symbol_diffs %>% readr::write_tsv(file.path(results_dir, output_file))
+  gene_symbol_diffs
+}
+print(paste("Probes not matching all methylation matrices =", length(rownames(probe_id_diffs))))
+```
+
+#### Check DNA SNV, CNV and SV biospecimen IDs
+
+- CNVkit seg file (cnv-cnvkit.seg.gz)
+```{r}
+# check that all biospecimen IDs are in the file
+biospecimen_ids <- readr::read_tsv(file.path(data_dir, "cnv-cnvkit.seg.gz")) %>%
+  dplyr::pull(ID) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing"),
+                is.na(RNA_library),
+                sample_type == "Tumor",
+                cohort != "DGD") %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-cnv-cnvkit.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("DNA biospecimen in histolgies missing in cnv-cnvkit.seg.gz = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "cnv-cnvkit.seg-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("DNA biospecimen in cnv-cnvkit.seg.gz missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- Control-FREEC file (cnv-controlfreec.tsv.gz)
+```{r}
+# check that all  biospecimen IDs are in the file
+biospecimen_ids <- readr::read_tsv(file.path(data_dir, "cnv-controlfreec.tsv.gz")) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing"),
+                is.na(RNA_library),
+                sample_type == "Tumor",
```

Comment:
```suggestion
```
remove these - since now some `sample_type` may be missing, we should just use something like `!is.na(pathology_diagnosis)` to designate non-normals.